base/sparse/sparsematrix.jl

```diff
@@ -193,6 +193,20 @@ end
 copy(S::SparseMatrixCSC) =
     SparseMatrixCSC(S.m, S.n, copy(S.colptr), copy(S.rowval), copy(S.nzval))
 
+function copy!{TvA, TiA, TvB, TiB}(A::SparseMatrixCSC{TvA,TiA},
+                                   B::SparseMatrixCSC{TvB,TiB})
+    if size(A) == size(B)
+        copy!(A.colptr, B.colptr)
+        resize!(A.nzval, length(B.nzval))
+        resize!(A.rowval, length(B.rowval))
+        copy!(A.rowval, B.rowval)
+        copy!(A.nzval, B.nzval)
+    else
+        invoke(Base.copy!, Tuple{AbstractMatrix{TvA}, AbstractMatrix{TvB}}, A, B)
```

Comment:
You can also give offsets to both arrays

```jl
julia> a = rand(5); b = rand(3);

julia> copy!(a, 2, b, 2);

julia> a[2:3] == b[2:end]
true
```

I don't think it is too uncommon to want to copy a chunk of an array into some other place and for now I think `copy!` is the most efficient.